### ISSUE TITLE
Couple issues when generating Chat preview4

### ISSUE THREAD

**Issue by knvsl** [Azure/autorest.swift#305]
1. `AddChatParticipantsErrors`, `CreateChatThreadErrors`, `AddChatParticipantsResult`, `CreateChatThreadResult` does not conform to Equatable.

```
/// Errors encountered during the addition of the chat participant to the chat thread.
public struct AddChatParticipantsErrors: Codable, Equatable {
    // MARK: Properties
```

- Generated code does not implement the equal operator or should not be equatable, in this case I don't think it makes sense to have to be Equatable
- All the structs with the same issue are using CommunicationError in some way
- Fix for now is to remove Equatable

2. In `Enumerations.swift`, `CommunicationCloudEnvironmentModel` uses public as one of its cases, causing error `Keyword public cannot be used an identifier here`

```
/// The cloud that the identifier belongs to.
public enum CommunicationCloudEnvironmentModel: RequestStringConvertible, Codable, Equatable {
    /// Custom value for unrecognized enum values
    case custom(String)

    case public
```

- Fix is to wrap public in backticks

**Comment by tjprescott** [comment]
At least for the second issue, that has already been fixed in PR #300. Update autorest.swift to the latest version and regenerate.

**Comment by tjprescott** [comment]
> At least for the second issue, that has already been fixed in PR #300. Update autorest.swift to the latest version and regenerate.

Nope. Discovered the edge case that is causing this to still fail. I'll create a fix.
